### PR TITLE
Provide a method to find graph elements by their UUID

### PR DESCRIPTION
**Pull request by phschaad**
- Provide a utility function to receive a unique identifier for graph elements.
- Provide a utility function to find graph elements based on their unique identifier.

### CODE REVIEW

File: sdfg_utils.js [hunk 1]
```diff
@@ -1,5 +1,137 @@
 // Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
 
+/**
+ * Return the string UUID for an SDFG graph element.
+ *
+ * UUIDs have the form of G\S\N\E, where:
```
Comment (tbennun) [inline sdfg_utils.js:6]:
```suggestion
 * UUIDs have the form of "G/S/N/E", where:
```

File: sdfg_utils.js [hunk 2]
```diff
@@ -1,5 +1,137 @@
...
+    return (
+        undefined_val + '/' +
+        undefined_val + '/' +
+        undefined_val + '/' +
+        undefined_val
+    );
+}
+
+function recursively_find_graph(graph, graph_id, ns_node=undefined) {
+    if (graph.node(0).sdfg.sdfg_list_id === graph_id) {
```
Comment (tbennun) [inline sdfg_utils.js:50]: strange :)

File: sdfg_utils.js [hunk 3]
```diff
@@ -1,5 +1,137 @@
...
+        const found_graph = recursively_find_graph(graph, graph_id);
+        graph = found_graph.graph;
+        result = {
+            parent: graph,
+            element: found_graph.node,
+        };
+    }
+
+    let state = undefined;
+    if (state_id > -1 && graph !== undefined) {
```
Comment (tbennun) [inline sdfg_utils.js:104]: `!== -1`?

File: sdfg_utils.js [hunk 4]
```diff
@@ -1,5 +1,137 @@
...
+        state = graph.node(state_id);
+        result = {
+            parent: graph,
+            element: state,
+        };
+    }
+
+    if (node_id > -1 && state !== undefined) {
```
Comment (tbennun) [inline sdfg_utils.js:112]: same

File: sdfg_utils.js [hunk 5]
```diff
@@ -1,5 +1,137 @@
...
+        // Look for a node in a state.
+        result = {
+            parent: state.data.graph,
+            element: state.data.graph.node(node_id),
+        };
+    } else if (edge_id > -1 && state !== undefined) {
```
Comment (tbennun) [inline sdfg_utils.js:118]: same

File: sdfg_utils.js [hunk 6]
```diff
@@ -1,5 +1,137 @@
...
+        // Look for an edge in a state.
+        result = {
+            parent: state.data.graph,
+            element: state.data.graph.edge(edge_id),
+        };
+    } else if (edge_id > -1 && state === undefined) {
```
Comment (tbennun) [inline sdfg_utils.js:124]: same

**Review by tbennun — COMMENTED** [review]
Minor edits and questions

File: sdfg_utils.js (hunk 2 above)
Comment (phschaad) [reply sdfg_utils.js:50]: Is there a prettier way to do it? Can't seem to see a more direct way..

**Review by tbennun — APPROVED** [review]

**Pull request merged**